modules/acm_customer/src/Plugin/rest/resource/CustomerDeleteResource.php

```diff
@@ -36,8 +95,8 @@ class CustomerDeleteResource extends ResourceBase {
    *   HTTP Response.
    */
   public function post(array $data = []) {
-    // If 'email' key is not available.
-    if (!$data['email']) {
+
+    if (!array_key_exists("email",$data)) {
       $this->logger->error('Invalid data to delete customer.');
       $response['success'] = (bool) (FALSE);
```

Comment:
this false here is making ACM retry it infinite times, are we doing anything to avoid it?

also are we sure we are sending email here? I see lot of logs with missing email and it seems not working as of now on current live sites.

Reply:
Regarding `success = false`, I'm sure that is the correct response. Regarding the problem of Commerce Connector always retrying false, that is a different problem and I propose we create a new ticket to address that: Perhaps a dead message queue or such; and perhaps we need to change the `success` response to have three values: `true`, `retry`, `never` whereby a `never` response would indicate to Commerce Connector that it should not retry and instead should send to the dead message queue and notify someone.

Regarding sending of the email address, we are certainly sending it, but **without** these two newly added functions, when `post($data)` is called `$data = []`.

Reply:
about the change in ACM, I think your suggestion makes sense

we could then send retry for deadlock issues

I don't have any comment for actual change here, since PR title said fixing I was thinking it will fix the infinite retry issue, which I can see is not the case.

Reply:
I don't get what is the actual behavioural change with this diff